nsiqcppstyle_types.py

```diff
@@ -1,69 +1,45 @@
 # Copyright (c) 2022 All rights reserved.
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions are
-# met:
-#
-#    * Redistributions of source code must retain the above copyright
-# notice, this list of conditions and the following disclaimer.
-#    * Redistributions in binary form must reproduce the above
-# copyright notice, this list of conditions and the following disclaimer
-# in the documentation and/or other materials provided with the
-# distribution.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# SPDX-License-Identifier: BSD-3-Clause
 
 from typing import NewType, Union
 import nsiqcppstyle_lexer
 import nsiqcppstyle_checker
 
 # A stack showing the lexical context the token resides in
-ContextStackType = NewType('ContextStackType', nsiqcppstyle_checker.ContextStack)
+ContextStack = NewType('ContextStack', nsiqcppstyle_checker.ContextStack)
 
 # The new context entry that will be pushed to the contextStack due to this token
-ContextType = NewType('ContextType', nsiqcppstyle_checker.Context)
+Context = NewType('Context', nsiqcppstyle_checker.Context)
 
 # A boolean flag denoting whether the function is a declaration (True) or definition (False)
-DeclarationType = NewType('DeclarationType', bool)
+Declaration = NewType('Declaration', bool)
 
 # A directory name
-DirNameType = NewType('DirNameType', str)
+DirName = NewType('DirName', str)
 
 # The name of the file being analyzed
-FileNameType = NewType('FileNameType', str)
+FileName = NewType('FileName', str)
 
 # A full function name
-FullNameType = NewType('FullNameType', str)
+FullFunctionName = NewType('FullFunctionName', str)
 
 # The lexer object used to analyze the source file
-LexerType = NewType('LexerType', nsiqcppstyle_checker.CppLexerNavigator)
+Lexer = NewType('Lexer', nsiqcppstyle_checker.CppLexerNavigator)
```

Comment:
Maybe TypeAlias here? We're renaming the class as Lexer for typing reasons, but not forcing an explicit conversion

Reply:
`TypeAlias` was introduced in Python 3.10.  There may be users that have not upgraded to 3.10.  I believe we should play it safe in the meantime.

Reply:
I can't use TypeAlias in my Python 3.7.0.

Awaiting your call or to resolve this conversation.

Reply:
Debian Buster is the sole mainline release which doesn't support Python 3.8, and its security support ends this month. I'm pretty confident we can drop support for 3.7

Reply:
But TypeAliaz was introduced in Python 3.10 as [PEP 613](https://docs.python.org/3.10/library/typing.html#relevant-peps), so we should wait until support for 3.9 is a fact.

Reply:
Sure. That makes sense